web/src/components/apps/AppDetailPage.tsx

```diff
@@ -95,72 +92,58 @@ function AppDetailPage(props: Props) {
 
   const history = useHistory();
   const params = useParams<KotsParams>();
-  const [currentApp, setCurrentApp] = useState(
-    props.appsList?.find((w) => w.slug === params.slug)
-  );
-  const theme = useTheme();
-
-  const toggleDisplayRequiredKotsUpdateModal = (message: string) => {
-    setState({
-      displayRequiredKotsUpdateModal: !state.displayRequiredKotsUpdateModal,
-      requiredKotsUpdateMessage: message,
-    });
-  };
-
-  const toggleIsBundleUploading = (isUploading: boolean) => {
-    setState({ isBundleUploading: isUploading });
-  };
-
-  const toggleErrorModal = () => {
-    setState({ displayErrorModal: !state.displayErrorModal });
-  };
-
-  const getApp = async (slug = params.slug) => {
-    if (!slug) {
-      return;
-    }
+  const { selectedApp } = useSelectedApp();
+  const {
+    data: appsData,
+    error: appsError,
+    isError: appsIsError,
+    isLoading: appsIsLoading,
+    refetch: refetchApps,
+  } = useApps();
 
-    try {
-      setState({ loadingApp: true });
+  const { apps: appsList } = appsData || {};
 
-      const res = await fetch(`${process.env.API_ENDPOINT}/app/${slug}`, {
-        headers: {
-          Authorization: Utilities.getToken(),
-          "Content-Type": "application/json",
-        },
-        method: "GET",
+  // loading state stuff that was in the old getApp() implementation
+  useEffect(() => {
```

Comment:
removed `getApp()` and replaced all of the state setting it does with this effect hook. The status code is handled in the fetch function so the handling logic here has been altered to just show the error message that comes from the fetch function. 

The old implementation would look at the status code and report it in the error message